lib/foreman/telemetry_sinks/statsd_sink.rb

```diff
@@ -26,23 +26,38 @@ def add_histogram(name, description, instance_labels, buckets)
 
       def increment_counter(name, value, tags)
         if @protocol == :datadog
-          StatsD.increment(name, value, tags: tags_in_statsd(tags))
+          if tags&.any?
+            # https://github.com/Shopify/statsd-instrument/issues/298
+            StatsD.increment(name, value, tags: tags_in_statsd(tags))
+          else
+            StatsD.increment(name, value)
+          end
         else
           StatsD.increment(name_tag_mapping(name, tags), value)
         end
       end
 
       def set_gauge(name, value, tags)
         if @protocol == :datadog
-          StatsD.gauge(name, value, tags: tags_in_statsd(tags))
+          if tags&.any?
+            # https://github.com/Shopify/statsd-instrument/issues/298
+            StatsD.gauge(name, value, tags: tags_in_statsd(tags))
+          else
+            StatsD.gauge(name, value)
+          end
         else
           StatsD.gauge(name_tag_mapping(name, tags), value)
         end
       end
 
       def observe_histogram(name, value, tags)
         if @protocol == :datadog
-          StatsD.measure(name, value, tags: tags_in_statsd(tags))
+          if tags&.any?
+            # https://github.com/Shopify/statsd-instrument/issues/298
+            StatsD.measure(name, value, tags: tags_in_statsd(tags))
+          else
+            StatsD.measure(name, value)
+          end
```

Comment:
These three changes are actually a blocker for Foreman 3.1 - there is a bug in the older version of statsd library we use in 3.1 that is causing issues in the PCP PMDA. I am going to bump the version in nightly, but not 3.1 that train is out of the station now.